tools/osv-scanner/internal/osv/osv.go

```diff
@@ -57,25 +64,60 @@ func MakePURLRequest(purl string) *Query {
 	}
 }
 
-func MakeRequest(request BatchedQuery) (*BatchedResponse, error) {
-	requestBytes, err := json.Marshal(request)
-	if err != nil {
-		return nil, err
+func MakePkgDetailsRequest(pkgDetails lockfile.PackageDetails) *Query {
+	return &Query{
+		Version: pkgDetails.Version,
+		Commit:  pkgDetails.Commit,
+		Package: Package{
+			Name:      pkgDetails.Name,
+			Ecosystem: string(pkgDetails.Ecosystem),
+		},
 	}
-	requestBuf := bytes.NewBuffer(requestBytes)
+}
 
-	resp, err := http.Post(QueryEndpoint, "application/json", requestBuf)
-	if err != nil {
-		return nil, err
+// From: https://stackoverflow.com/a/72408490
+func chunkBy[T any](items []T, chunkSize int) [][]T {
+	var _chunks = make([][]T, 0, (len(items)/chunkSize)+1)
+	for chunkSize < len(items) {
+		items, _chunks = items[chunkSize:], append(_chunks, items[0:chunkSize:chunkSize])
 	}
-	defer resp.Body.Close()
+	return append(_chunks, items)
+}
+
+func MakeRequest(request BatchedQuery) (*BatchedResponse, error) {
+	// API has a limit of 1000 bulk query per request
+	queryChunks := chunkBy(request.Queries, 990)
```

Comment:
nit: can you put 990 in a global const, and explain why this isn't 1000?

Reply:
Well it probably can be 1000, just wanted to avoid potential edge cases so used 990. Changed it to 1000 in the constant now.